crates/store/src/state.rs

```diff
@@ -698,45 +703,80 @@ impl State {
         // because changing one of them would lead to inconsistent state.
         let inner_state = self.inner.read().await;
 
+        let account_ids: Vec<AccountId> =
+            account_requests.iter().map(|req| req.account_id).collect();
+
         let state_headers = if !include_headers {
             BTreeMap::<AccountId, AccountStateHeader>::default()
         } else {
             let infos = self.db.select_accounts_by_ids(account_ids.clone()).await?;
-
             if account_ids.len() > infos.len() {
                 let found_ids = infos.iter().map(|info| info.summary.account_id).collect();
                 return Err(DatabaseError::AccountsNotFoundInDb(
                     BTreeSet::from_iter(account_ids).difference(&found_ids).copied().collect(),
                 ));
             }
 
-            infos
-                .into_iter()
-                .filter_map(|info| {
-                    info.details.map(|details| {
-                        (
-                            info.summary.account_id,
-                            AccountStateHeader {
-                                header: Some(AccountHeader::from(&details).into()),
-                                storage_header: details.storage().get_header().to_bytes(),
-                                // Only include account code if the request did not contain it
-                                // (known by the caller)
-                                account_code: request_code_commitments
-                                    .contains(&details.code().commitment())
-                                    .not()
-                                    .then_some(details.code().to_bytes()),
-                            },
-                        )
-                    })
-                })
-                .collect()
+            let mut headers_map = BTreeMap::new();
+
+            for account_info in infos {
```

Comment:
I would probably iterate over the requests instead? Since you also do that in the storage map loop.

Reply:
Changed it!